### PR TITLE
feat: fees modal

### PR DESCRIPTION
**Pull request by Xaroz**
This PR is taken from https://github.com/hyperlane-xyz/hyperlane-warp-ui-template/pull/746, removing only the piece related to tokenFeeQuote and getting lowest fee route

- Adds FeeSectionButton and TransferFeeModal that displays the amount of transfer fee the user has to pay
- Update useFeeQuotes to reflect the changes related to fees
- Update hyperlane packages

## Summary by CodeRabbit

* **New Features**
  * Added a fee section button and a modal showing detailed local and interchain fee breakdowns.
  * Added utilities to compute and consolidate transfer fees and to select the optimal transfer token.

* **Updates**
  * Updated Hyperlane registry, SDK, utils, and widgets to newer versions.
  * Added a documentation link for fee estimation.
  * Review/details view updated to conditionally surface fee information and improved fee-fetching behavior (optional highest-collateral token search).

* **Tests**
  * Added comprehensive tests covering fee aggregation and edge cases.

### CODE REVIEW

File: src/features/transfer/TransferTokenForm.tsx [hunk 1]
```diff
@@ -635,91 +636,115 @@ function ReviewDetails({
...
     amountWei,
-    visible,
+    isReview,
+  );
+  const { isLoading: isQuoteLoading, fees: feeQuotes } = useFeeQuotes(
+    values,
+    true,
+    originToken,
+    !isReview,
   );
```
[the remark this answers is not in the document]
Comment (Xaroz) [reply src/features/transfer/TransferTokenForm.tsx:652]: if originToken changes, tokenIndex will always change so no need to add it dependencies

File: src/features/transfer/useFeeQuotes.ts [hunk 2]
```diff
@@ -1,52 +1,75 @@
...
+): Promise<WarpCoreFeeEstimate | null> {
   if (!destination || !sender || !originToken) return null;
+  let transferToken = originToken;
+
+  if (searchForHighestCollateralToken) {
+    const destinationToken = originToken.getConnectionForChain(destination)?.token;
+    if (destinationToken) {
+      transferToken = await getTransferToken(warpCore, originToken, destinationToken);
+    }
+  }
```
[the remark this answers is not in the document]
Comment (Xaroz) [reply src/features/transfer/useFeeQuotes.ts:68]: `getTransferToken` already has a try...catch and will always return a token

File: src/features/transfer/TransferTokenForm.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (Xaroz) [reply src/features/transfer/TransferTokenForm.tsx:652]: the routeOverrideToken is only used when the form is inReview which has its own fee logic, if its not inReview it will trigger `searchHighestCollateral` and retrieve the override. could probably use some refactoring
